### PR TITLE
[Autotuner] CI Smoke Test - Sample & Iteration

### PR DESCRIPTION
**Pull request by luarss**
Targets argument sample and iteration

### CODE REVIEW

File: flow/test/test_helper.sh [hunk 1]
```diff
@@ -63,4 +63,40 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+  pip install -r tools/AutoTuner/requirements.txt
+
+  PLATFORM=${PLATFORM//-/}
+  PLATFORM=${PLATFORM^^}  
+
+  #echo "Running Autotuner smoke tune test"
+  #python3 -m unittest tools.AutoTuner.test.smoke_test_tune.${PLATFORM^^}TuneSmokeTest.test_tune
+  
+  #echo "Running Autotuner smoke sweep test"
+  #python3 -m unittest tools.AutoTuner.test.smoke_test_sweep.${PLATFORM^^}SweepSmokeTest.test_sweep
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]: Remove code instead of commenting it out.

File: tools/AutoTuner/test/smoke_test_sample_iteration.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# Disable GF180 tests for now.
+# class ASAP7SampleIterationSmokeTest(BaseSampleIterationSmokeTest):
+#     platform = "gf180"
+#     design = "gcd"
+# 
+#     def test_sample_iteration(self):
+#         for command in self.commands:
+#             out = subprocess.run(command, shell=True, check=True)
+#             successful = out.returncode == 0
+#             self.assertTrue(successful)
```
Comment (vvbandeira) [inline tools/AutoTuner/test/smoke_test_sample_iteration.py]: Remove code instead of commenting it out.

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: flow/test/test_helper.sh [hunk 3]
```diff
@@ -63,4 +63,34 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+    RUN_AUTOTUNER=1
+    ;;
+esac
+if [ $RUN_AUTOTUNER -eq 1 ]; then
+  case $PLATFORM in
+       "gf180" | "nangate45" | "sky130hd_fakestack" | "sky130hs")
+        RUN_AUTOTUNER=0
+        ;;
+  esac
+fi
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]:
```suggestion
RUN_AUTOTUNER=0
if [ "$DESIGN_NAME" = "gcd" ]; then
  case "$PLATFORM" in
    "gf180" | "nangate45" | "sky130hd_fakestack" | "sky130hs")
      RUN_AUTOTUNER=0
      ;;
    *)
      RUN_AUTOTUNER=1
      ;;
  esac
fi
```

File: flow/test/test_helper.sh [hunk 4]
```diff
@@ -63,4 +63,34 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+
+if [ $RUN_AUTOTUNER -eq 1 ]; then
+  # change directory to ../
+  cd ..
+  echo "Install dependencies in Venv"
+  python3 -m venv .venv
+  source .venv/bin/activate
+  pip install -r tools/AutoTuner/requirements.txt
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]: `setup.sh` script I mentioned in the other PR would handle this part.

File: flow/test/test_helper.sh [hunk 5]
```diff
@@ -63,4 +63,34 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+
+  PLATFORM=${PLATFORM//-/}
+  PLATFORM=${PLATFORM^^}
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]: Add a comment explaining what these expressions do and why they are required.

File: flow/test/test_helper.sh [hunk 6]
```diff
@@ -63,4 +63,34 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+  cd ..
+  echo "Install dependencies in Venv"
+  python3 -m venv .venv
+  source .venv/bin/activate
+  pip install -r tools/AutoTuner/requirements.txt
+
+  PLATFORM=${PLATFORM//-/}
+  PLATFORM=${PLATFORM^^}  
+
+  echo "Running Autotuner smoke sample & iteration test"
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]: `[INFO FLW-xyz]`

File: flow/test/test_helper.sh [hunk 7]
```diff
@@ -63,4 +63,34 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+  python3 -m unittest tools.AutoTuner.test.smoke_test_sample_iteration.${PLATFORM^^}SampleIterationSmokeTest.test_sample_iteration
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]: Why do we need `${PLATFORM^^}` twice?

File: flow/test/test_helper.sh (hunk 4 above)
Comment (luarss) [reply flow/test/test_helper.sh]: Updated, please check.

**Review by vvbandeira — APPROVED** [review]

**Comment by vvbandeira** [comment]
Please resolve merge conflicts.

**Comment by vvbandeira** [comment]
Still has merge conflicts.

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**